pkg/metrics/roundtripper.go

```diff
@@ -0,0 +1,151 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/defaults"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// appRegistryRoundTripperCounter wraps a RoundTripperFunc with functions that record prometheus metrics
+// when reaching out to an AppRegistry.
+func appRegistryRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper) promhttp.RoundTripperFunc {
+	return promhttp.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		resp, err := next.RoundTrip(r)
+		if err == nil {
+			if quayNamespace, ok := r.URL.Query()["namespace"]; ok && len(quayNamespace) == 1 {
+				counter.With(labels(r.Method, resp.StatusCode, quayNamespace[0])).Inc()
+			}
+		}
+		return resp, err
+	})
+}
+
+// labels adds the expected labels for the AppRegistry Counter.
+func labels(reqMethod string, status int, quayNamespace string) prometheus.Labels {
+	labels := prometheus.Labels{}
+	labels["code"] = sanitizeCode(status)
+	labels["method"] = strings.ToLower(reqMethod)
+	labels["opsrc"] = getOpsrcLabelValue(quayNamespace)
+
+	return labels
+}
+
+// getOpsrcLabelValue returns the quayNamespace value if it
+// matches a default OperatorSource. Otherwise, a generic
+// name is returned to prevent leaking customer data to telemetry.
+func getOpsrcLabelValue(quayNamespace string) string {
+	if defaults.IsDefaultSource(quayNamespace) {
+		return quayNamespace
+	}
+	return "non-default-opsrc"
```

Comment:
I think this is a good idea:
- We don't want an explosion of `opsrc` tag values in telemetry. So `non-default-opsrc` tag basically represents all custom operators globally.
- We are probably not interested in queries/alerts that targets an end user's custom opsrc, are we?